test/unit/quantum/results/test_reckoning.py

```diff
@@ -131,30 +167,41 @@ class TestCanonicalReckoner:
     """Test CanonicalReckoner."""
 
     @mark.parametrize(
-        "counts, operators, expected",
+        "frequencies, operators, expected",
         [
             ([], [], ReckoningResult(0, 0)),
             ([Counts({})], ["Z"], ReckoningResult(0, 1)),
             ([Counts({0: 0})], ["I"], ReckoningResult(0, 1)),
-            ([Counts({0: 1})], ["I"], ReckoningResult(1, 0)),
+            ([QuasiDistribution({0: 1}, shots=1)], ["I"], ReckoningResult(1, 0)),
             ([Counts({1: 1})], ["I"], ReckoningResult(1, 0)),
-            ([Counts({0: 1})], ["Z"], ReckoningResult(1, 0)),
+            ([QuasiDistribution({0: 1}, shots=1)], ["Z"], ReckoningResult(1, 0)),
             ([Counts({1: 1})], ["Z"], ReckoningResult(-1, 0)),
-            ([Counts({0: 1})], ["X"], ReckoningResult(1, 0)),
+            ([QuasiDistribution({0: 1}, shots=1)], ["X"], ReckoningResult(1, 0)),
             ([Counts({1: 1})], ["X"], ReckoningResult(-1, 0)),
-            ([Counts({0: 1})], ["Y"], ReckoningResult(1, 0)),
+            ([QuasiDistribution({0: 1}, shots=1)], ["Y"], ReckoningResult(1, 0)),
             ([Counts({1: 1})], ["Y"], ReckoningResult(-1, 0)),
             ([Counts({0: 1, 1: 1})], ["I"], ReckoningResult(1, 0)),
-            ([Counts({0: 1, 1: 1})], ["Z"], ReckoningResult(0, 1 / sqrt(2))),
+            (
+                [QuasiDistribution({0: 0.5, 1: 0.5}, shots=2)],
+                ["Z"],
+                ReckoningResult(0, 1 / sqrt(2)),
+            ),
             ([Counts({0: 1, 1: 1})], ["X"], ReckoningResult(0, 1 / sqrt(2))),
-            ([Counts({0: 1, 1: 1})], ["Y"], ReckoningResult(0, 1 / sqrt(2))),
+            (
+                [QuasiDistribution({0: 0.5, 1: 0.5}, shots=2)],
+                ["Y"],
+                ReckoningResult(0, 1 / sqrt(2)),
+            ),
             (
                 [Counts({0: 1, 1: 1}), Counts({0: 1, 1: 1})],
                 ["I", "Z"],
                 ReckoningResult(1, 1 / sqrt(2)),
             ),
             (
-                [Counts({0: 1, 1: 1}), Counts({0: 1, 1: 1})],
+                [
+                    QuasiDistribution({0: 0.5, 1: 0.5}, shots=2),
+                    QuasiDistribution({0: 0.5, 1: 0.5}, shots=2),
+                ],
```

Comment:
Don't delete tests, add new ones only; even if you know them as equivalent. 

They are indeed equivalent, but this is because of how you have implemented the code which you should not take into account when testing: i.e. _black box testing_.